packages/app/components/settings/tabs/billing.tsx

```diff
@@ -0,0 +1,266 @@
+  setPaymentByDefault,
+}: CreditCardItemProps) => {
+  const isDark = useIsDarkMode();
+  const { width } = useWindowDimensions();
+  const isMdWidth = width >= breakpoints["md"];
+  return (
+    <View tw="flex-row justify-between py-3.5">
+      <View tw="flex-row items-center justify-center">
+        <View tw="self-start md:self-center">
+          <CreditCard
+            width={isMdWidth ? 24 : 16}
+            height={isMdWidth ? 24 : 16}
+            color={isDark ? colors.white : colors.black}
+          />
+        </View>
+        <View tw="ml-2 flex-col items-start md:ml-0 md:flex-row md:items-center">
+          <Text tw="ml-0 text-sm font-medium text-gray-900 dark:text-white md:ml-2 md:text-base">
+            Ending in {data.details.last4}
+          </Text>
+          <View tw="mx-2 h-2 w-0 rounded-full bg-black dark:bg-white md:h-[2px] md:w-[2px]" />
+          <Text tw="text-sm font-medium text-gray-500 dark:text-gray-200 md:text-base">
+            Expiry
+            {` ${data.details.exp_year}/${data.details.exp_month}`}
+          </Text>
+        </View>
+      </View>
+      <View tw="flex-row">
+        <Button
+          size="small"
+          variant="tertiary"
+          style={data.is_default ? { backgroundColor: colors.lime[200] } : {}}
+          disabled={data.is_default}
+          onPress={setPaymentByDefault}
+        >
+          {data.is_default ? (
+            <>
+              <Check color={colors.gray[900]} />
+              <Text tw="ml-1 text-xs font-bold">Set as Default</Text>
+            </>
+          ) : (
+            "Set as Default"
+          )}
+        </Button>
+        <Button
+          size="small"
+          tw="ml-2"
+          iconOnly={!isMdWidth}
+          variant={isMdWidth ? "tertiary" : "danger"}
+          onPress={removePayment}
+        >
+          {isMdWidth ? "Remove" : <Trash />}
+        </Button>
+      </View>
+    </View>
+  );
+};
+const HistoryItem = memo(function HistoryItem({
+  item,
+}: {
+  item: PaymentsHistory;
+}) {
+  return (
+    <View tw="flex-row justify-between py-3.5 px-4 md:px-0">
+      <View tw="flex-col items-start justify-center md:flex-row md:items-center">
+        <Text tw="text-sm font-medium text-gray-900 dark:text-white md:text-base">
+          {item.amount}
+        </Text>
+        <View tw="h-2 w-0 md:w-6" />
+        <Text tw="text-sm font-medium text-gray-900 dark:text-white md:text-base">
+          {item.receipt_email}
+        </Text>
+        <View tw="h-2 w-0 md:w-6" />
+        <Text tw="text-sm font-medium text-gray-900 dark:text-white md:text-base">
+          {format(new Date(item.created_at), "MMM dd, Y")}
+        </Text>
+      </View>
+      <View tw="h-8 flex-row">
+        {item?.receipts?.length > 0 && (
+          <Button
+            variant="tertiary"
+            onPress={() => {
+              for (let link of item.receipts) {
+                Linking.openURL(link);
+              }
+            }}
```

Comment:
I'd rather pass a `useCallback` here and determine `link` in a `useMemo`

Reply:
yeah, absolutely can optimize a little bit here, but I think using `useCallback` and `useMemo` will make the code more complicated, WDTY?

Reply:
You might not need useCallback, but I would change the way you retrieve the link. Currently, this would, in theory trigger multiple `openURL` calls. If you want to map or filter, better to store as useMemo and pass as dep to a useCallback.

Reply:
Fixed